videostore/rawsegmenter.c

```diff
@@ -58,71 +47,63 @@ int video_store_raw_seg_init_h264(struct raw_seg_h264 **ppRS, // OUT
   codecCtx = avcodec_alloc_context3(codec);
   if (codecCtx == NULL) {
     av_log(NULL, AV_LOG_ERROR,
-           "video_store_raw_seg_init_h264 failed to allocate codec context");
+           "video_store_raw_seg_init failed to allocate codec context\n");
     ret = VIDEO_STORE_RAW_SEG_RESP_ERROR;
     goto cleanup;
   }
 
   codecCtx->width = width;
   codecCtx->height = height;
-  codecCtx->extradata = av_malloc(extradataSize + AV_INPUT_BUFFER_PADDING_SIZE);
-  if (codecCtx->extradata == NULL) {
-    av_log(NULL, AV_LOG_ERROR,
-           "video_store_raw_seg_init_h264 failed to allocate extradata");
-    ret = VIDEO_STORE_RAW_SEG_RESP_ERROR;
-    goto cleanup;
-  }
-  codecCtx->extradata_size = (int)extradataSize;
 
-  // memcpy can't fail
-  memcpy(codecCtx->extradata, extradata, extradataSize);
-  /* // Copy the codec parameters from the input stream to the output
-   * stream. This is equivalent */
-  /* // to -c:v copy in ffmpeg cli. This is needed to make sure we do not
-   * re-encode the stream. */
```

Comment:
this is no longer needed as the caller will now be expected to prepend sps, pps, and for h265 vps in the WritePacket call